sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/UtilsImpl.java

```diff
@@ -185,8 +187,29 @@ public static String computeDigest(ByteBuffer buffer) {
         }
     }
 
+    public static MessageDigest createSha256() {
+        try {
+            return MessageDigest.getInstance("SHA-256");
+
+        } catch (NoSuchAlgorithmException e) {
+            throw LOGGER.logExceptionAsError(new RuntimeException(e));
+        }
+    }
+
+    public static void validateDigest(MessageDigest messageDigest, String requestedDigest) {
+        String sha256 = byteArrayToHex(messageDigest.digest());
+        if (requestedDigest.length() != 71
+            || !requestedDigest.startsWith("sha256:")
+            || !requestedDigest.endsWith(sha256)) {
+            throw LOGGER.atError()
+                .addKeyValue("requestedDigest", requestedDigest)
+                .addKeyValue("actualDigest", () -> "sha256:" + sha256)
+                .log(new ServiceResponseException("The digest in the response does not match the expected digest."));
+        }
+    }
+
     private static final char[] HEX_ARRAY = "0123456789abcdef".toCharArray();
-    private static String byteArrayToHex(byte[] bytes) {
+    public static String byteArrayToHex(byte[] bytes) {
```

Comment:
We use this is a handful of places throughout the SDKs, mind filing an issue to make this a utility method in CoreUtils

Reply:
sure, #33405